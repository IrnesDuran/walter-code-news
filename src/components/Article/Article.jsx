import React from 'react';
import { withRouter } from 'react-router-dom';

const Article = (props) => {
    return(
        <div className="group flex flex-col bg-white m-4 transform hover:-translate-y-1 hover:shadow-md transition ease-in-out duration-100">
            <div className="flex-1 relative">
                <img className="object-cover w-full h-64 object-center" 
                    src={props.article.urlToImage ? props.article.urlToImage :"https://upload.wikimedia.org/wikipedia/commons/f/fc/No_picture_available.png"} alt="Article"/>
                <div className="absolute py-1 px-3 bg-white text-teal-500 uppercase font-extrabold text-sm      bottom-0 left-0 -my-4">{props.article.author}
                </div>
            </div>
            <div className="w-full h-64 p-4 flex flex-col justify-between">
                <h1 className="text-lg font-black">{props.article.title}</h1>
                <p className="text-sm">{props.article.description}</p>
                <button className="text-teal-500 cursor-pointer w-28 self-end hover:text-tertiary transform scale-0 group-hover:scale-100 transition ease-in-out duration-100"
                    onClick={() => props.history.push(`articles/${props.article.title}`)}>Read full article &rarr;</button>
            </div>
        </div>

    )};

export default withRouter(Article);

