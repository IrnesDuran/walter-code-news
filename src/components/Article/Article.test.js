import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import Article from './Article';

describe ("Article component test",()=> {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Article/>);

    });
    



    it('should test Article component snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    });

})