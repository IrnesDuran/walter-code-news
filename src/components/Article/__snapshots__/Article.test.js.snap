// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component test should test Article component snapshot 1`] = `ShallowWrapper {}`;
